src/Compilers/Core/Portable/Operations/Operation.cs

```diff
@@ -101,8 +125,8 @@ protected void SetParentOperation(IOperation parent)
             var result = Interlocked.CompareExchange(ref _parentDoNotAccessDirectly, parent, s_unset);
 
             // tree must belong to same semantic model if parent is given
-            Debug.Assert(parent == null || ((Operation)parent).SemanticModel == SemanticModel ||
-                ((Operation)parent).SemanticModel == null || SemanticModel == null);
+            Debug.Assert(parent == null || ((Operation)parent).OwningSemanticModel == OwningSemanticModel ||
+                ((Operation)parent).OwningSemanticModel == null || OwningSemanticModel == null);
```

Comment:
>(Operation)parent).OwningSemanticModel == null || OwningSemanticModel == null [](start = 17, length = 77)

Shouldn't this be an &&?

Reply:
Actually, shouldn't this never be hit at all? Both semantic models should always be equal, even if they're null.

---
In reply to: [197514109](https://github.com/dotnet/roslyn/pull/27922#discussion_r197514109) [](ancestors = 197514109)

Reply:
> Actually, shouldn't this never be hit at all? Both semantic models should always be equal, even if they're null.

This is pre-existing logic. It is quite possible it is obsolete, but I am not sure.

---
In reply to: [197514479](https://github.com/dotnet/roslyn/pull/27922#discussion_r197514479) [](ancestors = 197514479,197514109)

Reply:
Yes, I just did a rename. We can probably try cleaning/investigating this in a separate PR.

---
In reply to: [197523553](https://github.com/dotnet/roslyn/pull/27922#discussion_r197523553) [](ancestors = 197523553,197514479,197514109)